third_party_libraries/kml.js

```diff
@@ -660,6 +648,62 @@ L.Util.extend(L.KML, {
 		return coords;
 	},
 
+
+	/**
+	 * Gets the style options for the placemarks leaflet marker
+	 *
+	 * @param {xml Element} place A placemark XML element
+	 * @param {path} style Style options for the placemark
+	 * @return {path[]} options The style options for the placemark
+	 */
+	getOptions: function (place, style) {
+		var i, j, el, options = {};
+		el = place.getElementsByTagName('styleUrl');
+
+		for (i = 0; i < el.length; i++) {
+			var url = el[i].childNodes[0].nodeValue;
+			for (var a in style[url])
+			{
+				options[a] = style[url][a];
+			}
+		}
+
+		return options;
+	},
+
+	/**
+	 * Creates a base layergroup element for the placemarks leaflet icon
+	 *
+	 * @param {xml Element} place A placemark XML element
+	 * @param {path[]} options The style options for the placemark
+	 * @param {xml} xml Xml data related to the placemark
+	 * @return {FeatureGroup} layer Basic location and style data pertaining to
+	 * 		the placemark
+	 */
+	initializeLayer: function (place, options, xml) {
+		var layers = [];
+		var parse = ['LineString', 'Polygon', 'Point'];
+		for (j in parse) {
+			var tag = parse[j];
+			el = place.getElementsByTagName(tag);
+
+			for (i = 0; i < el.length; i++) {
+				var l = this["parse" + tag](el[i], xml, options);
```

Comment:
Without tracing too deeply, this will attempt(?) to run `parseLineString`/`parsePolygon`/`parsePoint` on the elements in `el`.  Since we know the value of `tag`, we could probably just do a `switch`/`case` based on that and only call the appropriate method.

Alternatively, we could just unroll the loop and just sequentially calling the different 'types' (whatever `LineString` et al are)